Light edits to `&is_used` and `&is_assigned` (#159)
zeek/zeek-docs: script-reference/attributes.rst

&is_used
--------

Zeek has static analysis capabilities for detecting locations in a script where
local variables are assigned values that are not subsequently used (i.e. "dead
code").

It can also warn about unused functions, hooks, and event handlers.  The intent
behind these checks is to catch instances where the script writer has introduced
typos in names, or has forgotten to remove code that's no longer needed.  For

> functions and hooks, "unused" means the function/hook is not exported or in the

Reviewer: ```suggestion
functions and hooks, "unused" means the function/hook is neither exported nor in the
```
I think this works with the (nor deprecated) removed.

global scope (nor deprecated), and no "live" (i.e., not "unused")
function/hook/event handler calls it.  For event handlers, "unused" means that
the event engine does not generate the event, nor do any "live"

> function/hook/event handler generates (and the event handler is not deprecated).

Reviewer: ```suggestion
function/hook/event handler generate it (and the event handler is not deprecated).
```

Maybe move the deprecated special-case mentions out of parenthesis and into a separate sentence in the beginning or end?

"Functions/hooks/event handlers marked deprecated (:zeek:see:`&deprecated`) are never reported unused."

Author: Great suggestion, yeah — that paragraph is hairy. :-) Forthcoming in a force-push.
